builtin/providers/azure/resource_azure_hosted_service.go

```diff
@@ -0,0 +1,190 @@
+	location := d.Get("location").(string)
+	reverseDNS := d.Get("reverse_dns_fqdn").(string)
+	description := d.Get("description").(string)
+
+	// set the label as the resource's ID:
+	label := getRandomStringLabel(20)
+	d.SetId(label)
+
+	err := hostedServiceClient.CreateHostedService(
+		hostedservice.CreateHostedServiceParameters{
+			ServiceName:    serviceName,
+			Location:       location,
+			Label:          label,
+			Description:    description,
+			ReverseDNSFqdn: reverseDNS,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("Failed defining new Azure hosted service: %s", err)
+	}
+
+	return nil
+}
+
+// resourceAzureHostedServiceExists checks whether a hosted service with the
+// given service_name already exists on Azure.
+func resourceAzureHostedServiceExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	azureClient := meta.(*AzureClient)
+	hostedServiceClient := hostedservice.NewClient(azureClient.managementClient)
+
+	log.Println("[INFO] Querying for hosted service existence.")
+	serviceName := d.Get("service_name").(string)
+	_, err := hostedServiceClient.GetHostedService(serviceName)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
+			// it means that the hosted service has been deleted in the meantime...
+			d.SetId("")
+			return false, nil
+		} else {
+			return false, fmt.Errorf("Failed to query for hosted service name availability: %s", err)
+		}
+	}
+
+	return true, nil
+}
+
+// resourceAzureHostedServiceRead does all the necessary API calls
+// to read the state of a hosted service from Azure.
+func resourceAzureHostedServiceRead(d *schema.ResourceData, meta interface{}) error {
+	azureClient := meta.(*AzureClient)
+	hostedServiceClient := hostedservice.NewClient(azureClient.managementClient)
+
+	log.Println("[INFO] Querying for hosted service info.")
+	serviceName := d.Get("service_name").(string)
+	hostedService, err := hostedServiceClient.GetHostedService(serviceName)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
+			// it means the hosted service was deleted in the meantime,
+			// so we must remove it here:
+			d.SetId("")
+			return nil
+		} else {
+			return fmt.Errorf("Failed to get hosted service: %s", err)
+		}
+	}
+
+	log.Println("[DEBUG] Reading hosted service query result data.")
+	d.Set("service_name", hostedService.ServiceName)
+	d.Set("url", hostedService.URL)
+	d.Set("location", hostedService.Location)
+	d.SetId(hostedService.Label)
+	d.Set("description", hostedService.Description)
+	d.Set("status", hostedService.Status)
+	d.Set("reverse_dns_fqdn", hostedService.ReverseDNSFqdn)
+	d.Set("default_certificate_thumbprint", hostedService.DefaultWinRmCertificateThumbprint)
+
+	return nil
+}
+
+// resourceAzureHostedServiceUpdate does all the necessary API calls to
+// update some settings of a hosted service on Azure.
+func resourceAzureHostedServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
```

Comment:
No need to include the blank stub method here. If there's no implementation, just leave `Update` as `nil` in the resource struct.